compiler/il/OMRILOps.cpp

```diff
@@ -291,3 +297,30 @@ OMR::ILOpCode::compareOpCode(TR::DataType dt,
    return TR::BadILOp;
    }
 
+/**
+ * \brief
+ *    Return a comparison type given the compare opcode.
+ *
+ * \parm op
+ *    The compare opcode.
+ *
+ * \return
+ *    The compareison type.
+ */
+TR_ComparisonTypes
+OMR::ILOpCode::getCompareType(TR::ILOpCodes op)
+   {
+   if (isStrictlyLessThanCmp(op))
+      return TR_cmpLT;
+   else if (isStrictlyGreaterThanCmp(op))
+      return TR_cmpGT;
+   else if (isLessCmp(op))
+      return TR_cmpLE;
+   else if (isGreaterCmp(op))
+      return TR_cmpGE;
+   else if (isEqualCmp(op))
+      return TR_cmpEQ;
+   else if (isNotEqualCmp(op))
+      return TR_cmpNE;
+   TR_ASSERT(false, "Expect comparison opcode");
```

Comment:
This fallthrough path should either return a value, or fail to return entirely. Otherwise, when assertions are disabled, it will return garbage that's unlikely to be a valid `TR_ComparisonTypes`.

One way to avoid returning would be to use `TR_ASSERT_FATAL` instead of `TR_ASSERT`. Although I'm not sure whether that would be clear to compilers. It would be good to avoid compiler warnings of "not all paths return a value."